### PR TITLE
ui: Switch to an icon font for UI icons

### PR DESCRIPTION
**Pull request by special**
Icon fonts handle scaling, positioning, and high DPI much better than
image icons do, and can be considerably smaller. No attempt is made
here to improve on the current UI icons, but one should be.

The font is generated by Fontello, and can be reproduced and modified
using the included .json file.

I haven't tested this on Windows. It shouldn't be merged until that happens.

### CODE REVIEW

File: icons/LICENSE [hunk 1]
```diff
@@ -5,3 +5,41 @@ This work is licensed under the Creative Commons Attribution-ShareAlike 4.0
...
 http://creativecommons.org/licenses/by-sa/4.0/ or send a letter to
 Creative Commons, PO Box 1866, Mountain View, CA 94042, USA.
+
+
+## Font Awesome
+
+   Copyright (C) 2012 by Dave Gandy
+
+   Author:    Dave Gandy
+   License:   SIL ()
```
Comment (rburchell) [inline icons/LICENSE]: Missing the URL present on the other instances.

File: src/ui/qml/MainToolBar.qml [hunk 2]
```diff
@@ -63,9 +79,9 @@ ToolBar {
 
         ToolButton {
             action: preferencesAction
-            iconSource: "qrc:/ui/icons/gear.png"
             implicitHeight: 24
-            implicitWidth: 24
+            style: iconButtonStyle
+            text: "\ue803"
```
Comment (rburchell) [inline src/ui/qml/MainToolBar.qml]: a "// this is X" at the end might not go astray on these, for those of us that aren't unicode-brained. sure, it's somewhat obvious if you read the context, but for scanning, it helps.

**Comment by rburchell** [comment]
Only real concern is about the font licensing. I'm not familiar with that license (and while on holiday, not able to research).

File: icons/LICENSE (hunk 1 above)
Comment (special) [inline icons/LICENSE]: This list is autogenerated by the nifty tool that creates a custom minimal font.

**Comment by special** [comment]
License is very permissive; in this case, I believe it doesn't restrict redistribution or even require attribution. Including the license text here is sufficient.

See also, https://github.com/fontello/fontello/wiki/What-about-license%3F

**Comment by special** [comment]
Tested and works on Windows too.

**Comment by rburchell** [comment]
:+1:

**Pull request merged**